### PR TITLE
JUCX: HelloWorld example:

### PR DESCRIPTION
**Pull request by netanelyo**
- Create server Worker and client Worker.
- Extract from each its Worker address.
- Exchange addresses through TCP socket.
- Example unit-test - will be executed during build.
- Added Options class - command-line arguments parser, to be used in example code.
- README: added README.md as symlink and added execution instructions to run the example code from command-line.

### CODE REVIEW

File: bindings/java/scripts/hello_world.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+TOP_DIR="$(cd "$(dirname "$0")"/../"" && pwd)"
+
+# Check arguments
+if [ -z $1 ]; then
+    echo -e "$0 ERROR: Missig first parameter. Should be 'server' or 'client'."
+    exit 1
+fi
+
+classpath=.:../jucx-0.0.1-SNAPSHOT.jar
```
Comment (miked-mellanox) [inline bindings/java/scripts/hello_world.sh]: Please autodetect jar file name as version can differ, i.e. ```jarf=$(find .. -name jucx-*SNAPSHOT,jar)```

File: bindings/java/src/main/java/org/ucx/jucx/util/Options.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
```
Comment (miked-mellanox) [inline bindings/java/src/main/java/org/ucx/jucx/util/Options.java]: s/2017/2018 in all files (and other java PRs)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6149/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6151/ for details (Mellanox internal link).

**Comment by netanelyo** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6156/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6281/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6320/ for details (Mellanox internal link).

**Pull request closed without merging**